Bottom: fd856c0ce4f708032ba5a33272d759bbd390e2bd
Top:    e2ee858c6b13a0140148edac8957f66ebf0103d1
Author: Alexey Lyashkov <shadow@Alexeys-MacBook-Pro.local>
Date:   2016-01-21 10:12:50 +0300

attach control message to IB

use IB verbs api to attach control messages
to incomming ring.
Send ctl mgs on wire to verify recv code works



---

diff --git a/ib-sock-ctl.c b/ib-sock-ctl.c
index 4f35dcd..f334383 100644
--- a/ib-sock-ctl.c
+++ b/ib-sock-ctl.c
@@ -42,6 +42,15 @@ ib_sock_ctl_take(struct IB_SOCK *sock)
  */
 int ib_sock_ctl_post(struct IB_SOCK *sock, struct ib_sock_ctl *msg)
 {
+	/* base on ib_mad_post_receive_mads() - that wr isn't need to live any time 
+	 * so - save memory and declare on stack.
+	 */
+	struct ib_recv_wr wr;
+	struct ib_recv_wr *bad_wr;
+	int ret;
+
+	ret = ib_post_recv(sock->is_qp, &wr, &bad_wr);
+
 	return 0;
 }
